tools/topology/topology2/include/components/gain.conf

```diff
@@ -152,6 +152,41 @@ Class.Widget."gain" {
 				}
 			}
 		}
+
+		# mute switch control
+		mixer."2" {
+			Object.Base.channel.1 {
+				name	"flw"
+				reg	2
+				shift	0
+			}
+			Object.Base.channel.2 {
+				name	"fl"
+				reg	2
+				shift	1
+			}
+			Object.Base.channel.3 {
+				name	"fr"
+				reg	2
+				shift	2
+			}
+			Object.Base.channel.4 {
+				name	"frw"
+				reg	2
+				shift	3
+			}
```

Comment:
I also have no understanding of what this is.

Are you trying to enable a per-channel mute?

IIRC the mute button and the LED applies to all channels, no?

Reply:
It's from the SWITCH_CHANNEL_MAP defined in tools/topology/topology1/platform/intel/intel-generic-dmic-m4 but I believe SOF Linux driver do not use the reg/shift values to control the switch.

You could also find same definitions in tools/topology/topology2/include/components/volume.conf but I'm told that IPC4 does not use this conf file. I guess the writer just translate the definitions from topology1 to topology2.

In IPC3, mute is per channel but in UCM it should applies to all channels.